app/components-react/pages/BrowseOverlays.tsx

```diff
@@ -85,11 +95,65 @@ export default function BrowseOverlays(p: {
     }
   }
 
+  /**
+   * Get a list of scenes in the active scene collection
+   *
+   * @returns An array of scenes with items including only `id` and `name`
+   */
+  async function getScenes() {
+    return ScenesService.views.scenes.map(scene => ({
+      id: scene.id,
+      name: scene.name,
+    }));
+  }
+
+  /**
+   * Adds a collectible to a scene.
+   *
+   * Collectibles are just a CDN URL of an image or video, this API provides
+   * embedded pages with a convenience method for creating sources based on those.
+   *
+   * @param sceneId - ID of the scene where the collectible will be added to
+   * @param assetURL - CDN URL of the collectible asset
+   * @param type - Type of source that will be created, `image` or `video`
+   *
+   * @returns string - ID of the scene item that was created for the source.
+   * @throws When type is not image or video.
+   * @throws When URL is a not a Streamlabs CDN URL.
+   * @throws When scene for the provided scene ID can't be found.
+   * @throws When it fails to create the source.
+   */
+  async function addCollectibleToScene(sceneId: string, assetURL: string, type: 'image' | 'video') {
+    if (!['image', 'video'].includes(type)) {
+      throw new Error("Unsupported type. Use 'image' or 'video'");
+    }
+
+    if (!assetURL.startsWith('https://cdn.streamlabs.com/marketplace')) {
+      throw new Error('Invalid asset URL');
+    }
+
+    const sourceSettings =
+      type === 'video' ? { is_local_file: false, input: assetURL } : { file: assetURL };
+
+    // TODO: find or create enum
+    const sourceType = type === 'video' ? 'ffmpeg_source' : 'image_source';
+
+    // TODO: do we want the caller to provide name?
+    const sourceName = SourcesService.views.suggestName('Collectible');
```

Comment:
I think the asset will probably have a name in the library, so maybe we should open that up as an argument?

Reply:
Commenting again since it got lost: we eventually ended up needing to pass from caller yes.